lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -317,7 +317,7 @@ def digital_wallet_payment_method?(payment_method)
       end
 
       def adding_network_token_card_data?(payment_method)
-        return true if payment_method.is_a?(ActiveMerchant::Billing::NetworkTokenizationCreditCard) && payment_method.source == :network_token
+        return true if payment_method.is_a?(ActiveMerchant::Billing::NetworkTokenizationCreditCard)
 
         false
       end
```

Comment:
What do you think about updating this method to have the check for `options.dig(:stored_credential, :initiator) != 'merchant'` as well so that the `add_network_token_cryptogram_and_eci` method can just rely on this one?

Reply:
Yes, done!